src/Build/BackEnd/BuildManager/BuildManager.cs

```diff
@@ -176,6 +176,11 @@ public class BuildManager : INodePacketHandler, IBuildComponentHost, IDisposable
         /// </summary>
         private int _nextBuildSubmissionId;
 
+        /// <summary>
+        /// The last BuildParameters used for building.
+        /// </summary>
+        private BuildParameters _previousBuildParameters = null;
```

Comment:
Why keep the whole `BuildParameters` rooted when you only need the priority bit?

Reply:
I was going to do something a bit more complicated but retracted a lot of it when I backed out the more invasive task host changes. Will switch to a bit.